### PR TITLE
Replace name and friendly name in full adopted configs

### PR DESCRIPTION
**Pull request by jesserockz**
# What does this implement/fix?

This fixes an issue where setting `name_add_mac_suffix: true` and `import_full_config: true` together results in a local config after adoption without the known mac suffix.

This does direct string replacements on the hardcoded name and friendly name to update them with the mac suffix value.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/dashboard_import/__init__.py [hunk 1]
```diff
@@ -82,56 +81,48 @@ def import_config(
...
+        try:
+            req = requests.get(url, timeout=30)
+            req.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            raise ValueError(f"Error while fetching {url}: {e}") from e
+
+        contents = req.text
+
+        # remove mac suffix (-######) to get original name
+        original_name = name[:-7]
```
Comment (balloob) [inline esphome/components/dashboard_import/__init__.py]: shouldn't you parse it first to make sure it had a Mac suffix? Or is that mandatory for import url?

Comment (jesserockz) [reply esphome/components/dashboard_import/__init__.py]: Good point, its not mandatory

**Comment by nielsnl68** [comment]
What is the status of this PR?

**Comment by jesserockz** [comment]
I still need to finish this PR off

**Comment by pszafer** [comment]
@jesserockz do you plan to work on that in nearby future? I'm not pushing, just asking, because if no, then I'd like to make some PR to esphome creators docs, that `import_full_config: true` doesn't allow `creators` device to be successfully imported by end user. I' ve spent whole weekend trying to get it working until I started to look in source code ;)

**Comment by jesserockz** [comment]
@pszafer Thanks for pushing me. I have made some more updates to the PR that does the replacement a bit cleaner.

**Pull request merged**